crates/flux-infer/src/infer.rs

```diff
@@ -516,6 +516,8 @@ impl<'genv, 'tcx> InferCtxtAt<'_, '_, 'genv, 'tcx> {
                 this.subtyping(actual, formal, reason)?;
             }
 
+            // Check invariants
+            // TODO(struct-inv): check-invariants-here?
```

Comment:
yes, we should also check the invariants here. Otherwise, I think we would fail to catch the following


```rust
use flux_rs::attrs::*;

#[refined_by(n: int)]
#[invariant(n > 0)])
struct A { 
    #[field(n)]
    f: i32 
}

fn baz(a: A) {
    flux_rs::assert(a.f > 0); // uh-oh
}

fn foo() {
    let mut a = A { f : 1 };
    a.f -= 1;  // this is fine, we don't need to preserve the invariant while unfolded.
    baz(a);   // force folding
}
```